### PR TITLE
Batch crafting

### PR DESCRIPTION
**Pull request by PlinHost**
## About The Pull Request
It's back.
img
The plus button prompts user for amount.
img

## Changelog
:cl:
add: Added batch crafting to craft menu for certain recipes
/:cl:

### CODE REVIEW

File: code/datums/craft/recipe.dm [hunk 1]
```diff
@@ -67,6 +67,52 @@
 	var/datum/craft_step/CS = steps[step]
 	return CS.apply(I, user, target, src)
 
+/datum/craft_recipe/proc/build_batch(mob/living/user, amount)
+	if(steps.len > 1)
+		warning("Attempted to batch craft a multi-step recipe: [name]")
```
Comment (ITCups) [inline code/datums/craft/recipe.dm]: do we really need this unnecessary logging ?

File: code/datums/craft/recipe.dm [hunk 2]
```diff
@@ -67,6 +67,52 @@
 	var/datum/craft_step/CS = steps[step]
 	return CS.apply(I, user, target, src)
 
+/datum/craft_recipe/proc/build_batch(mob/living/user, amount)
```
Comment (ITCups) [inline code/datums/craft/recipe.dm]: make this proc utilize try_build() instead of copypasting code

Comment (PlinHost) [reply code/datums/craft/recipe.dm]: That would result in lots of stacks containing only one item spawning when the result is a stack. This proc puts all items in one stack in that case.

File: code/datums/craft/recipe.dm (hunk 1 above)
Comment (PlinHost) [reply code/datums/craft/recipe.dm]: User asks for fifty but gets one because some coder messed up a recipe? Yeah why should we log that

Comment (ITCups) [reply code/datums/craft/recipe.dm]: mention that then

File: code/datums/craft/recipe.dm (hunk 2 above)
Comment (ITCups) [reply code/datums/craft/recipe.dm]: make try_build() return result if succeeded and stack all stackables in build_batch()

Comment (PlinHost) [reply code/datums/craft/recipe.dm]: And where should the spawn code from try_build go?

Comment (ITCups) [reply code/datums/craft/recipe.dm]: it stays there, you just merge all stackables that try_build() returned

File: code/datums/craft/recipe.dm (hunk 1 above)
Comment (PlinHost) [reply code/datums/craft/recipe.dm]:
```suggestion
		warning("A multi-step recipe has BATCH_CRAFT flag: [name]. It should not!")
```
Like that?

File: code/datums/craft/recipe.dm (hunk 2 above)
Comment (PlinHost) [reply code/datums/craft/recipe.dm]: resolved in 54bced1e

File: code/datums/craft/recipe.dm (hunk 1 above)
Comment (ITCups) [reply code/datums/craft/recipe.dm]: yeah good enough

**Pull request merged**